### PR TITLE
[RecoveryServices.Backup] Adding data source move feature - Copy-AzRecoveryServicesVault cmdlet

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Adding data move feature for moving data sources across vaults. This is a high priority feature for region decommissioning and will be released to customers as an OOB release.

Copy-AzRecoveryServicesVault cmdlet: copies the data from source vault to target vault

**Cmdlet review PR:**
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/677

## Checklist

- [Yes] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [Yes] The title of the PR is clear and informative
- [Yes] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [Yes] If applicable, the changes made in the PR have proper test coverage
- [Yes] For public API changes to cmdlets:
    - [Yes] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [Yes] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@hiaga Please fix the errors in generating help.

File: src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.ps1 [hunk 1]
```diff
@@ -12,6 +12,32 @@
...
 
+function Test-AzureBackupDataMove
+{
+	$sourceLocation = "centraluseuap"
+	$sourceResourceGroup = Create-ResourceGroup $sourceLocation 7
+
+	$targetLocation = "eastus2euap"
+	$targetResourceGroup = Create-ResourceGroup $targetLocation 8
+		
+	$vm = Create-VM "hiagaSrcRG2" $sourceLocation 3
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.ps1:23]: create a random name instead of using hardcoded one

File: src/RecoveryServices/RecoveryServices.Backup/Helpers/BackupUtils.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            string DataSourceType,
+            ServiceClientAdapter serviceClientAdapter)
+        {
+                ODataQuery<ProtectedItemQueryObject> queryParams = new ODataQuery<ProtectedItemQueryObject>(
+                                                                    q => q.BackupManagementType
+                                                                            == BackupManagementType &&
+                                                                         q.ItemType == DataSourceType);
+
+                List<ProtectedItemResource> protectedItems = new List<ProtectedItemResource>();
+                string skipToken = null;
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices.Backup/Helpers/BackupUtils.cs:41]: can be removed

File: src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md [hunk 3]
```diff
@@ -182,14 +185,17 @@ Creates an ASR storage classification mapping in the Recovery Services vault.
 ### [New-AzRecoveryServicesAsrvCenter](New-AzRecoveryServicesAsrvCenter.md)
 Adds a vCenter server to discover protectable items from.
 
+### [New-AzRecoveryServicesAsrVMNicConfig](New-AzRecoveryServicesAsrVMNicConfig.md)
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md:188]: Why is ASR changes part of this PR?

**Comment by hiaga** [comment]
@sambitratha resolved review comments and PR check issues. Please review.

**Comment by hiaga** [comment]
@msjingli resolved the errors. Please review

File: src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md [hunk 4]
```diff
@@ -275,15 +281,15 @@ Starts the test failover cleanup operation.
 ### [Start-AzRecoveryServicesAsrTestFailoverJob](Start-AzRecoveryServicesAsrTestFailoverJob.md)
 Starts a test failover operation.
 
-### [Start-AzRecoveryServicesAsrUnplannedFailoverJob](Start-AzRecoveryServicesAsrUnplannedFailoverJob.md)
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md:278]: Revert this change

**Review by sambitratha — APPROVED** [review]

**Comment by hiaga** [comment]
@msJinLei  Adding the reference cmdlet review PR: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/677

Please review and approve this DS move feature PR urgently. This is targeted for an Out of Band release today.

File: src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md [hunk 5]
```diff
@@ -30,7 +30,7 @@ Dismounts all the files of the recovery point.
 Edits a Site Recovery plan.
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Az.RecoveryServices.md:30]: Copy-AzRecoveryServicesVault not in the Az.RecoveryServices.md

File: src/RecoveryServices/RecoveryServices/ChangeLog.md [hunk 6]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Azure Backup added a new cmdlet Copy-AzRecoveryServicesVault for DS move feature.
+* Get-AzRecoveryServicesBackupJob cmdlet now supports operation type 'BackupDataMove'.
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/ChangeLog.md:22]: Please regenerate Get-AzRecoveryServicesBackupJob.md to include the modification

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by hiaga** [comment]
@msJinLei Adding the needed changes.

**Review by msJinLei — APPROVED** [review]
